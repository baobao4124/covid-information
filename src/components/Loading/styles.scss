@import "../../assets/css/variables";

//Loading
.loading{ position: absolute; top: 0; bottom: 0; left: 0; right: 0;
  .loading-icon{ background-image: $brand-icon5; width: 25px; height: 25px;}
}

.processing{ display: flex; align-items: center;
  .processing-name{ margin-left: 20px;}
}

.skype-loader{ width: 60px; height: 60px; position: relative;
  .dot{ position: absolute; top: 0; left: 0; width: 60px; height: 60px; animation: 1.7s dotrotate cubic-bezier(0.775, 0.005, 0.310, 1.000) infinite;
    &:after, .first{ background-color: #666; border-radius: 50%; content: ""; position: absolute; width: 8px; height: 8px; left: 50%; margin-left: -4px;}
    &:nth-child(1){ animation-delay: 0.2s;}
    &:nth-child(2){ animation-delay: 0.35s;}
    &:nth-child(3){ animation-delay: 0.45s;}
    &:nth-child(4){ animation-delay: 0.55s;}
    .first{ background-color: #666; margin-top: -4px; animation: 1.7s dotscale cubic-bezier(0.775, 0.005, 0.310, 1.000) infinite; animation-delay: 0s;}
  }
}

@keyframes dotrotate {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

@keyframes dotscale {
  0%, 10% { width: 16px; height: 16px; margin-left: -8px;  margin-top: -4px; }
  50% { width: 16px; height: 16px; margin-left: -8px; margin-top: -4px; }
  90%, 100% { width: 16px; height: 16px;  margin-left: -8px;  margin-top: -4px; }
}














