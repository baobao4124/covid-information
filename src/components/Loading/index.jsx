// Libraries
import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

const defaultProps = {
    top: 0,
    marginTop: 0,
    isShowLoading: true,
    loading: {},
    loadingIcon: {}
};

const Loading = (props) => {
    return (
        <div className="loading" style={{top: props.top, paddingTop: props.paddingTop, zIndex: 1005, ...props.loading}}>
            <div className="loading-icon" style={{marginTop: props.marginTop, marginLeft: 'auto', marginRight: 'auto', ...props.loadingIcon}} />
            {
                props.isShowLoading && <div className="text-center font-weight-bold">Loading ...</div>
            }
        </div>
    );
};

Loading.propTypes = {
    top: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    marginTop: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    isShowLoading: PropTypes.bool,
    loading: PropTypes.object,
    loadingIcon: PropTypes.object
};

Loading.defaultProps = defaultProps;

export default Loading;
