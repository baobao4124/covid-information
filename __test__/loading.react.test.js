'use strict';

import React from 'react';
import Loading from 'Src/components/Loading';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

it('renders correctly', () => {
    const component = renderer.create(<Loading top={10} />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

test('Testing render correct style', () => {
    const loadingStyle = {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        top: 0,
        paddingTop: '10px',
        zIndex: 1000
    };

    const loadingIcon = {
        marginTop: '-60px',
        marginLeft: 'auto',
        marginRight: 'auto'
    };
    
    let component = shallow(<Loading loading={loadingStyle} />);

    expect(component.find('.loading .text-center').text()).toEqual('Loading ...');
    expect(component.find('.loading').props().style).toEqual(loadingStyle);

    component.setProps({isShowLoading: false});
    
    expect(component.find('.loading .text-center')).toEqual({});
    
    component.setProps(loadingIcon);
    
    expect(component.find('.loading > .loading-icon').props().style).toEqual(loadingIcon);
});
