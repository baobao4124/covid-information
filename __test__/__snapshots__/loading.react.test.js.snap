// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="loading"
  style={
    Object {
      "paddingTop": undefined,
      "top": 10,
      "zIndex": 1005,
    }
  }
>
  <div
    className="loading-icon"
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 0,
      }
    }
  />
  <div
    className="text-center font-weight-bold"
  >
    Loading ...
  </div>
</div>
`;
